com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs

```diff
@@ -58,27 +72,117 @@ private void GetTransparencySortingMode(Camera camera, ref SortingSettings sorti
-        {
-            // start checking at the next layer
-            for (var i = startLayerIndex+1; i < sortingLayers.Length; i++)
+            // Draw renderers
+            using (new ProfilingScope(cmd, m_ProfilingDrawRenderers))
             {
-                if(!CompareLightsInLayer(startLayerIndex, i, sortingLayers))
-                    return i-1;
+                CoreUtils.SetRenderTarget(cmd, colorAttachment, depthAttachment, ClearFlag.None, Color.white);
+
+                for (var i = 0; i < batchSize; i++)
+                {
+                    using (new ProfilingScope(cmd, m_ProfilingDrawLayerBatch))
+                    {
+                        var layerBatch = layerBatches[startIndex + i];
+
+                        if (layerBatch.lightStats.totalLights > 0)
+                        {
+                            unsafe
+                            {
+                                for (var blendStyleIndex = 0; blendStyleIndex < blendStylesCount; blendStyleIndex++)
+                                {
+                                    var blendStyleMask = (uint)(1 << blendStyleIndex);
+                                    var blendStyleUsed = (layerBatch.lightStats.blendStylesUsed & blendStyleMask) > 0;
+
+                                    if (blendStyleUsed)
+                                        cmd.SetGlobalTexture(k_ShapeLightTextureIDs[blendStyleIndex], new RenderTargetIdentifier(layerBatch.renderTargetIds[blendStyleIndex]));
+
+                                    RendererLighting.EnableBlendStyle(cmd, blendStyleIndex, blendStyleUsed);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            for (var blendStyleIndex = 0; blendStyleIndex < k_ShapeLightTextureIDs.Length; blendStyleIndex++)
+                            {
+                                cmd.SetGlobalTexture(k_ShapeLightTextureIDs[blendStyleIndex], Texture2D.blackTexture);
+                                RendererLighting.EnableBlendStyle(cmd, blendStyleIndex, blendStyleIndex == 0);
+                            }
+                        }
+
+                        context.ExecuteCommandBuffer(cmd);
+                        cmd.Clear();
+
+                        filterSettings.sortingLayerRange = layerBatch.layerRange;
+                        context.DrawRenderers(renderingData.cullResults, ref drawSettings, ref filterSettings);
+
+                        // Immediately release light buffer textures after we're done using them, so that the temporary RTs can be re-used for the next batches.
+                        unsafe
+                        {
+                            for (var blendStyleIndex = 0; blendStyleIndex < blendStylesCount; blendStyleIndex++)
+                            {
+                                cmd.ReleaseTemporaryRT(layerBatch.renderTargetIds[blendStyleIndex]);
```

Comment:
can we find a way to ensure we don't create/release every batch?

Reply:
I no longer do this in the latest commits.